cmake/platform/arch/x86.cmake

```diff
@@ -0,0 +1,26 @@
+##############################################################################
+#
+# (c) Copyright IBM Corp. 2017
+#
+#  This program and the accompanying materials are made available
+#  under the terms of the Eclipse Public License v1.0 and
+#  Apache License v2.0 which accompanies this distribution.
+#
+#      The Eclipse Public License is available at
+#      http://www.eclipse.org/legal/epl-v10.html
+#
+#      The Apache License v2.0 is available at
+#      http://www.opensource.org/licenses/apache2.0.php
+#
+# Contributors:
+#    Multiple authors (IBM Corp.) - initial implementation and documentation
+###############################################################################
+
+set(OMR_ARCH_DEFINITIONS "")
+
+if(OMR_ENV_DATA64)
+   list(APPEND OMR_ARCH_DEFINITIONS J9HAMMER)
+else()
+   list(APPEND OMR_ARCH_DEFINITIONS J9X86)
+endif()
```

Comment:
I think the overlap between the two is sufficiently large that I'd like to keep them in one file unless we have some evidence that we should split them out. Previous experience in compiler builds indicated they were substantially the same.